crates/engine/tree/src/tree/root.rs

```diff
@@ -342,21 +359,14 @@ where
 
         // Dispatch proof gathering for this state update
         scope.spawn(move |_| {
-            let provider = match view.provider_ro() {
-                Ok(provider) => provider,
-                Err(error) => {
-                    error!(target: "engine::root", ?error, "Could not get provider");
-                    return;
-                }
-            };
-
-            // TODO: replace with parallel proof
-            let result = Proof::overlay_multiproof(
-                provider.tx_ref(),
-                // TODO(alexey): this clone can be expensive, we should avoid it
-                input.as_ref().clone(),
-                proof_targets.clone(),
-            );
+            let result = ParallelProof::new(
+                config.consistent_view.clone(),
+                config.nodes_sorted.clone(),
+                config.state_sorted.clone(),
+                config.prefix_sets.clone(),
+            )
```

Comment:
https://github.com/paradigmxyz/reth/pull/13135

```suggestion
                .with_branch_node_hash_masks(true)
            )
```